### PR TITLE
Raise OSError if on Windows before Python 3.2.

### PR DESCRIPTION
**Pull request by jmcgeheeiv**
Support for links was added to Windows in Python 3.2.  When attempting to create a symbolic or hard link on an earlier Python version, raise an exception.

@mrbean-bremen, please review.

### CODE REVIEW

File: pyfakefs/fake_filesystem.py [hunk 1]
```diff
@@ -1427,7 +1429,10 @@ def CreateLink(self, file_path, link_target):
 
     Raises:
       IOError:  if the file already exists
+      OSError:  if on Windows before Python 3.2
     """
+    if not _is_link_supported:
+        raise OSError("Symbolic links are not supported on Windows before Python 3.2")
     resolved_file_path = self.ResolvePath(file_path)
```
Comment (mrbean-bremen) [inline pyfakefs/fake_filesystem.py]: Use 2 spaces like in the rest of the file (or change it all to 4 spaces to follow pep-8, which I would really like!)

**Comment by mrbean-bremen** [comment]
@jmcgeheeiv : I added a PR with the skipped tests to this branch, so you may merge it first before merging this PR.

**Comment by jmcgeheeiv** [comment]
Yes, @mrbean-bremen, since `_is_link_allowed` is always `True` for me on Linux, I would like to ask your assistance getting Windows errors cleared up.

I should not push this to master, as it would break master on Windows.  Could you pull branch raise-on-unsupported-windows-links, get that working on Windows?  As an experiment on GitHub usage, I would be interested to see if now that you are a committer, you are allowed to push back to `origin/raise-on-unsupported-windows-links`.

**Comment by jmcgeheeiv** [comment]
Although this PR is a funny place for such a message, I have no other place to say it:

Thank you for all your solid contributions to pyfakefs, @mrbean-bremen.  Please join me as a committer on this project.

**Comment by mrbean-bremen** [comment]
Thank you for trusting me with this - I gladly join you as a committer (technically already did, and yes, I could merge the PR to your branch). 
For a little background how I came here - I needed a file system mock for a django project I did at work some time ago, and coming from C++, wrote one similar to a mock I once wrote for in C++/Qt ()in goggletest). This was not pythonic at all, and lately with a newer django version I ran into some initialization difficulties, which I discussed with a previous collegue - and he advised me to use pyfakefs, which I did. So my primary concern was to replace my own crude code with pyfakefs, using specifically the unittest part you wrote. Most of my contributions really have been to get this running under Windows, and add stuff I already had like the file system space.
Currently I'm not working on that project at all (doing C# stuff at the moment and a little Python in my free time), but got used to this library and just want it to be used by more people.
So basically, yes, I like to contribute more where needed, feel free to ask for any Windows-specific (or other) stuff. And yes, that's a funny place for this, but I think there is no dedicated place for discussions outside of issues and PRs in GitHub, to my knowledge.
